rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize.cpp

```diff
@@ -54,10 +54,117 @@ void ResizeNode::create_node()
+    std::vector<double> scale(dim, 1);
+    std::vector<bool> has_size(dim, false);
+    unsigned sizes_provided = 0;
+    bool has_max_size = _max_roi_size.size() > 0;
+    for (unsigned i = 0; i < dim; i++) {
+        has_size[i] = (_src_roi_size[i] != 0) && (_dst_roi_size[i] != 0);
+        sizes_provided += has_size[i];
+        scale[i] = _src_roi_size[i] ? (_dst_roi_size[i] / static_cast<double>(_src_roi_size[i])) : 1;
+    } if (_scaling_mode == RocalResizeScalingMode::ROCAL_SCALING_MODE_STRETCH) {
+        if (sizes_provided < dim) {
+            for (unsigned i = 0; i < dim; i++) {
+                if (!has_size[i])
+                    _dst_roi_size[i] = _src_roi_size[i];
+            }
+        }
+        if (has_max_size) {
+            for (unsigned i = 0; i < dim; i++) {
+                if ((_max_roi_size[i] > 0) && (_dst_roi_size[i] > _max_roi_size[i]))
+                    _dst_roi_size[i] = _max_roi_size[i];
+            }
+        }
+    }
+    else if (_scaling_mode == RocalResizeScalingMode::ROCAL_SCALING_MODE_DEFAULT) {
+        if (sizes_provided < dim) {
+            double average_scale = 1;
+            for (unsigned i = 0; i < dim; i++) {
+                if (has_size[i])
+                    average_scale *= scale[i];
+            }
+            if (sizes_provided > 1)
+                average_scale = std::pow(average_scale, 1.0 / sizes_provided);
+            for(unsigned i = 0; i < dim; i++) {
+                if(!has_size[i])
+                    _dst_roi_size[i] = std::round(_src_roi_size[i] * average_scale);
+            }
+        }
+        if (has_max_size) {
+            for (unsigned i = 0; i < dim; i++) {
+                if ((_max_roi_size[i] > 0) && (_dst_roi_size[i] > _max_roi_size[i]))
+                    _dst_roi_size[i] = _max_roi_size[i];
+            }
+        }
+    } else {
+        double final_scale = 0;
+        bool first = true;
+        for (unsigned i = 0; i < dim; i++) {
+            if (has_size[i]) {
+                double s = scale[i];
+                if (first ||
+                    (_scaling_mode == RocalResizeScalingMode::ROCAL_SCALING_MODE_NOT_SMALLER && s > final_scale) ||
+                    (_scaling_mode == RocalResizeScalingMode::ROCAL_SCALING_MODE_NOT_LARGER && s < final_scale)) {
+                        final_scale = s;
+                    }
+                first = false;
+            }
+        }
+        if(has_max_size) {
+            for (unsigned i = 0; i < dim; i++) {
+                if(_max_roi_size[i] > 0) {
+                    double s = static_cast<double>(_max_roi_size[i]) / _src_roi_size[i];
+                    if (s < final_scale)
+                        final_scale = s;
+                }
+            }
+        }
+        for (unsigned i = 0; i < dim; i++) {
+            if(!has_size[i] || (scale[i] != final_scale))
+                _dst_roi_size[i] = std::round(_src_roi_size[i] * final_scale);
+        }
+    }
+}
```

Comment:
Add line at EOF

Reply:
There is an empty line present at the EOF. It shows that in my vscode editor. In Github the last empty line is not shown for all the files.